src/react/utils/index.js

```diff
@@ -40,6 +40,10 @@ export function formatDate(d) {
   return `${d.toDateString()} ${d.toTimeString().split(' ')[0]}`;
 }
 
+export function formatRoleSessionName(oidc) {
+  return `${oidc.user?.profile.email || 'zenko-ui'}`;
```

Comment:
Is fallbacking to `zenko-ui` when the email of the user is not defined something planed in any tech design (I'm not aware of the feature details) ? I'm just asking because if it is not the case IMO when the user profile is missing an email we should likely prompt him with a message saying so, like "Invalid profile, missing required email". In the current implementation an user without email will be prompted with a not authorized role modal.

Reply:
I'm not sure about this. I didn't see a requirement for the session name, but we agreed that defining it as the unique `email` identifier from the OIDC JWT claim was a good way to allow for dissociable log generation for each identity. Your comment is true though, we should not use a default value.

I propose to default to the (mandatory) `sub` claim value if the email is not present. This claim is also a unique identitier of identities within keycloak, although it is not as human-readable. I will make the needed changes so that we correctly reject the request if the `oidc.user.profile` object is not defined, otherwise we cann fallback to the `sub` claim. What do you think?

Something like that:
return `${oidc.user?.profile.email || ${oidc.user?.profile.sub}`;

Reply:
Switched to the `sub` claim. We do not need to verify that the `oidc.user.profile` exists because it is already done in the src/react/actions/auth.js` file (please correct me if I'm wrong), so I removed the function.
See https://github.com/scality/zenko-ui/pull/203/commits/adcacbb9e7853271fc72347398fc3dc0929f24f3